x/dsmr/storage.go

```diff
@@ -231,11 +230,11 @@ func (s *chunkStorage[T]) SetMin(updatedMin int64, saveChunks []ids.ID) error {
 // GatherChunkCerts provides a slice of chunk certificates to build
 // a chunk based block.
 // TODO: switch from returning random chunk certs to ordered by expiry
-func (s *chunkStorage[T]) GatherChunkCerts() []*ChunkCertificate {
+func (s *chunkStorage[T]) GatherChunkCerts() []*ChunkCertificate[T] {
 	s.lock.RLock()
 	defer s.lock.RUnlock()
 
-	chunkCerts := make([]*ChunkCertificate, 0, len(s.chunkMap))
+	chunkCerts := make([]*ChunkCertificate[T], 0, len(s.chunkMap))
```

Comment:
(unrelated to this PR) Can we remove the below nil check if the cert will always be populated when it's in the map?

Reply:
Yeah I'm not sure why this check exists here either. It looks like this should be safe to remove...